### PR TITLE
Audit 3

### PR DESCRIPTION
**Pull request by ash-burnt**

### CODE REVIEW

File: integration_tests/minimum_fee_test.go [hunk 1]
```diff
@@ -48,12 +48,6 @@ func TestXionMinimumFee(t *testing.T) {
 		&xiontypes.MsgSetPlatformPercentage{},
 		&xiontypes.MsgSend{},
 	)
-	//_, err = xion.FullNodes[0].ExecTx(ctx,
```
Comment (ash-burnt) [inline integration_tests/minimum_fee_test.go:51]: Let's remove this if we don't need it

File: x/globalfee/ante/fee.go [hunk 2]
```diff
@@ -85,7 +85,8 @@ func (mfd FeeDecorator) GetTxFeeRequired(ctx sdk.Context, tx sdk.FeeTx) (sdk.Coi
 
 	// Get local minimum-gas-prices
 	localFees := GetMinGasPrice(ctx, int64(tx.GetGas()))
-	return MaxCoins(globalFees, localFees), nil
+	return CombinedFeeRequirement(globalFees, localFees)
+	//return MaxCoins(globalFees, localFees), nil
```
Comment (ash-burnt) [inline x/globalfee/ante/fee.go:89]: same here

File: x/globalfee/querier.go [hunk 3]
```diff
@@ -31,7 +31,6 @@ func (g GrpcQuerier) Params(stdCtx context.Context, _ *types.QueryParamsRequest)
 	var maxTotalBypassMinFeeMsgGasUsage uint64
 	ctx := sdk.UnwrapSDKContext(stdCtx)
 
-	// todo: if return err if not exist?
```
Comment (ash-burnt) [inline x/globalfee/querier.go:34]: If this is an invalid state, we should add checks to make it impossible elsewhere

File: x/globalfee/ante/fee.go (hunk 2 above)
Comment (edjroz) [inline x/globalfee/ante/fee.go:89]:
```suggestion
```

File: x/globalfee/querier.go (hunk 3 above)
Comment (edjroz) [reply x/globalfee/querier.go:34]: The invalid state already has validation:

https://github.com/burnt-labs/xion/blob/3466e9c34a17d3d8c53d8441d9e241892b08573f/x/globalfee/types/genesis.go#L34-L40

It should not be possible to start the module without setting the params.

Comment (ash-burnt) [reply x/globalfee/querier.go:34]: In that case, we can remove the todo, yea?

File: integration_tests/minimum_fee_test.go (hunk 1 above)
Comment (ash-burnt) [reply integration_tests/minimum_fee_test.go:51]: Bump

**Review by edjroz — APPROVED** [review]

**Pull request merged**